app/Http/Controllers/Api/JobApiController.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use App\Models\JobPoster;
+use App\Models\Criteria;
+use App\Services\Validation\JobPosterValidator;
+use App\Http\Requests\UpdateJobPoster;
+
+class JobApiController extends Controller
+{
+    public function __construct()
+    {
+        // This applies the appropriate policy to each resource route
+        $this->authorizeResource(JobPoster::class, 'job');
+    }
+
+    /**
+     * Convert a job poster to the array expected by API requests,
+     * with all criteria,
+     * and with translation arrays in both languages.
+     *
+     * @param JobPoster $job
+     * @return mixed[]
+     */
+    private function jobToArray(JobPoster $job)
+    {
+        $criteria = Criteria::where('job_poster_id', $job->id)->get();
+        $criteriaTranslated = [];
+        foreach ($criteria as $criterion) {
+            // TODO: getTranslationsArray probably makes DB calls every loop. Find a way to profile & optimize.
```

Comment:
Fun fact, it doesn't! I took a peek at the DB queries and this seems to be fine.

Reply:
Yay!